test/units/TEST-13-NSPAWN.machined.sh

```diff
@@ -367,6 +367,17 @@ varlinkctl --more call /run/systemd/machine/io.systemd.Machine io.systemd.Machin
 (! varlinkctl --more call /run/systemd/machine/io.systemd.Machine io.systemd.Machine.List '{"name": ".host"}' | grep 'acquireUIDShift')
 varlinkctl --more call /run/systemd/machine/io.systemd.Machine io.systemd.Machine.List '{"name": ".host", "acquireMetadata": "yes"}' | grep 'UIDShift'
 
+# test io.systemd.Machine.Open
+varlinkctl call /run/systemd/machine/io.systemd.Machine io.systemd.Machine.Open '{"name": ".host", "mode": "tty"}'
+varlinkctl call /run/systemd/machine/io.systemd.Machine io.systemd.Machine.Open '{"name": ".host", "mode": "login"}'
+varlinkctl call /run/systemd/machine/io.systemd.Machine io.systemd.Machine.Open '{"name": ".host", "mode": "shell"}'
+
+# TODO(ikruglov): how to really test other modes?
```

Comment:
hmm, that's a good question, i am not sure how we could make varlinkctl spit out fds in any reasonable form that the shell could use them... shells always work the other way round: the shell allocates an fd, and pass it to other processes, but the shall cannot accept fds from other processes.

Reply:
ah, here's an idea: maybe we should introduce a new "--exec=" switch or so for varlinkctl. You pass a cmdline of your choice, and we'll chainload that cmdline after a successful method call, and pass the method reply body as JSON input as stdin (`acquire_data_fd()` is great for that) and then pass any fds we received via fd 3 and bove.

we could even set $LISTEN_FDS env var to the number of fds received, so that the invoked process could use sd_listen_fds() to see what it got passed....

This would then mean you could do:

```c
varlinkctl call /run/systemd/machine/io.systemd.Machine io.systemd.Machine.Open '{"name": ".host", "mode": "shell"}' --exec='echo foo >3 ; read x <3 ; test "$x" == foo'

or something like that. you get the gist.

Reply:
ok make sense. I'd put it in a separate PR though. Mind merging without it and I'll follow up?

Reply:
@poettering, I've been thinking about your suggestion. Is there a real-life case where this --exec option would be useful? I mean: adding something just to run a test is too much IMO. There should be a simpler option if it's only for testing.